src/listener/starboard/starAdd.ts

```diff
@@ -0,0 +1,118 @@
+import { Listener } from 'discord-akairo';
+import { guildConfigs } from '../../guild/config/guildConfigs';
+import { MessageReaction } from 'discord.js';
+import { TextChannel } from 'discord.js';
+import { User } from 'discord.js';
+import { getPermissionLevel } from '../../util/permission/getPermissionLevel';
+import { PermissionLevel } from '../../util/permission/permissionLevel';
+import { Starboard } from '../../structure/starboard/starboardMessage';
+import { StarModel } from '../../model/starboardMessage';
+import { MessageEmbed } from 'discord.js';
+import { emojiEquals, findImg } from '../../util/functions';
+
+export default class StarAddListener extends Listener {
+	constructor() {
+		super('starAdd', {
+			emitter: 'client',
+			event: 'messageReactionAdd',
+		});
+	}
+
+	async exec(reaction: MessageReaction, user: User) {
+		const msg = reaction.message;
+		if (!msg.guild) return;
+		const config = guildConfigs.get(msg.guild.id);
+		const starboardConfig = config?.features.starboard;
+
+		if (
+			!config ||
+			!starboardConfig ||
+			!starboardConfig.channel ||
+			!starboardConfig.triggerAmount ||
+			(starboardConfig.ignoredChannels &&
+				starboardConfig.ignoredChannels.includes(msg.channel.id))
+		)
+			return;
+
+		if (msg.author.id === this.client.user?.id) return;
+
+		const minStarboardPermTriggerLvl =
+			starboardConfig.minimumPermLevel || PermissionLevel.Everyone;
+
+		const emojiAddedByUser = reaction.emoji;
+		const starboardTriggerEmoji = starboardConfig.emoji
+			? emojiEquals(this.client, starboardConfig.emoji)
+			: '⭐';
+
+		if (!emojiEquals(emojiAddedByUser, starboardTriggerEmoji)) return;
+
+		const member = msg.guild.member(user)!;
+		const userPermissionLvl = getPermissionLevel(member, this.client);
+
+		if (userPermissionLvl === PermissionLevel.Muted)
+			return reaction.users.remove(user);
+
+		if (msg.author.id === user.id && emojiEquals(emojiAddedByUser, starboardTriggerEmoji)) return reaction.users.remove(user)
```

Comment:
Is there a reason why you don't include a semi-colon here?

Reply:
Might be helpful to add semi colons to prettier config

Reply:
Made a PR for that: #64